### ISSUE TITLE
Ability to report implicit string concatenation as error

### ISSUE THREAD

**Issue by giyyapan** [microsoft/pyright#478]
**Is your feature request related to a problem? Please describe.**
Implicit string concatenation (`"ab" "cd" == "abcd"`) is a tricky feature. 

It's not just because we barely use it (well, I never use it!), but it makes some typo errors (like missing comma in a string list) extremely hard to find. In my opinion, this feature just does more harm than good.

Especially when they are assigned like this:
```python
my_long_str_list = [
    # some comment
    "some_item_a",
    # some comment
    "some_item_b" # <-- missing a comma here
    # some comment
    "some_item_c"
]
```
This kind of error may cause serious runtime problems that very hard to debug.

**Describe the solution you'd like**
Add a pyrightconfig.json option like `reportImplicitStringConcatenation`, and report all implicit string concatenations as error when the option is set to `true`.

**Additional context**
There's even a rejected [PEP](https://www.python.org/dev/peps/pep-3126/) discussing removing this syntax!

**Comment by erictraut** [comment]
Good suggestion!

**Comment by erictraut** [comment]
This will be included in the next version of Pyright. I decided not to enable it by default in strict mode since usage of implicit string concatenation is legitimate in many cases.

**Comment by giyyapan** [comment]
Awesome!
Looking forward to trying it. Thanks!

**Comment by erictraut** [comment]
This support is now included in 1.1.19, which I just published.
